pages/people/index.tsx

```diff
@@ -1,101 +1,153 @@
+		return {
+			props: {
+				allPeople: [],
+				departmentTree: departmentRecordsToDepartmentTree([]),
+			},
+		}
 	}
 }
 
 export default function PeoplePage({
 	allPeople,
 	departmentTree,
 }: Props): React.ReactElement {
+	const router = useRouter()
 	const [searchingName, setSearchingName] = useState('')
 	const [hideNoPicture, setHideNoPicture] = useState(false)
-	const [filteredDepartments, setFilteredDepartments] = useState([])
-
-	const peopleFiltered = filterPeople(
-		allPeople,
-		searchingName,
-		hideNoPicture,
-		findChildrenDepartments(
-			departmentTree,
-			filteredDepartments[filteredDepartments.length - 1]?.id || []
-		)
-	)
+	const [filteredDepartments, setFilteredDepartments] = useState<
+		DepartmentNode[]
+	>([])
+	const [people, setPeople] = useState<PersonRecord[]>(allPeople)
+	const [loading, setLoading] = useState(false)
+
+	useEffect(() => {
+		if (!router.isReady) {
+			return
+		}
+
+		const { search, hideNoPicture: hideNoPictureParam } = router.query
+
+		if (search && typeof search === 'string' && search !== searchingName) {
+			setSearchingName(search)
+		}
+
+		const shouldHideNoPicture = hideNoPictureParam === 'true'
+		if (shouldHideNoPicture !== hideNoPicture) {
+			setHideNoPicture(shouldHideNoPicture)
+		}
+	}, [router.isReady, router.query])
+
+	useEffect(() => {
+		if (!router.isReady) {
+			return
+		}
+		const fetchPeople = async () => {
+			setLoading(true)
+			try {
+				const query = {}
+
+				if (searchingName.trim()) {
+					query['search'] = searchingName.trim()
+				}
+
+				if (hideNoPicture) {
+					query['hideNoPicture'] = 'true'
+				}
+
+				router.push(
+					{
+						pathname: router.pathname,
+						query,
+					},
+					undefined,
+					{ shallow: true }
+				)
+
+				let queryParam = ''
+				if (hideNoPicture) {
+					queryParam += '?hideNoPicture=true'
+				}
+
+				if (searchingName.trim()) {
+					queryParam += queryParam
+						? `&search=${encodeURIComponent(searchingName.trim())}`
+						: `?search=${encodeURIComponent(searchingName.trim())}`
+				}
+
+				const response = await fetch(`/api/hashicorp${queryParam}`)
+				const data = await response.json()
+				setPeople(data.results)
+			} catch (err) {
+				console.error('Error fetching people:', err)
+				setPeople([])
+			} finally {
+				setLoading(false)
+			}
+		}
+
+		const timeoutId = setTimeout(fetchPeople, 300)
+		return () => clearTimeout(timeoutId)
+	}, [searchingName, hideNoPicture, router.isReady])
 
 	const filteredDepartmentIds = filteredDepartments.reduce(
 		(acc: string[], department: DepartmentNode) => [...acc, department.id],
 		[]
 	)
 
-	// Sr. candidate TODO: Update URL based on search and department filters
-
 	return (
 		<main className="g-grid-container">
-			<div>
-				<div>
-					<h1>HashiCorp Humans</h1>
-					<span>Find a HashiCorp human</span>
-				</div>
+			<section className={style['search-container']}>
+				<h1>HashiCorp Humans</h1>
+				<h5>Find a HashiCorp human</h5>
```

Comment:
Is there a reason why we jumped straight from a `h1` to an `h5` here?

Reply:
Yes, this was a tradeoff I should have mentioned. I didn't do much styling but for this I wanted to convert that to an `h2` header for optimization but needed it to style it so that it's not so drastically bigger than in design. I didn't get a chance to style it so I just went with an `h5`.